test/functional/rpc_getgeneralinfo.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the getgeneralinfo RPC."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, try_rpc
+
+class GetGeneralInfoTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = False
+        self.num_nodes = 1
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_cli()
+
+    def run_test(self):
+        """Test getgeneralinfo."""
+        """Check if 'getgeneralinfo' is callable."""
+        try_rpc(None, None, self.nodes[0].getgeneralinfo, None, None)
+        """Check if 'getgeneralinfo' is idempotent (multiple requests return same data)."""
+        json1 = self.nodes[0].getgeneralinfo()
```

Comment:
Why not put this into a tuple

Reply:
I don't understand the context. However, I am not that experienced in python so it could be that my code is not following best-practices.

Reply:
As a general thought, I'd humbly suggest spending time looking at existing tests here, reviewing, practising writing PRs without opening them, and polishing PRs more before adding them to the stack of open PRs to review. Maybe I'm doing this wrong, but I have dozens of finished commits that I've written locally without opening PRs to (a) to learn the codebase, and (b) to maintain a ratio of 5-15 PRs reviewed or issues tested per PR opened, as the number of open issues and PRs keeps growing and review/testing is where resources are needed.

Reply:
Well, the best way to learn something is by doing it. Reading docs is fine, sure, but in the end you have to "say something" so others can hear you (and give you a proper critique). Docs don't correct you if you get something wrong. Anyway, I think it's best to close this PR, just to make the PR-mountain a bit smaller. This RPC call isn't that important anyway, I think.

Reply:
I have a lot to learn but these two articles based on the past ten months might be helpful:
- https://jonatack.github.io/articles/how-to-open-bitcoin-core-prs
- https://jonatack.github.io/articles/how-to-review-bitcoin-core-prs

Reply:
@jonatack Is this PR really that bad? I spent a few weeks looking at the codebase and tests and I would have raised a similar PR.

Reply:
> @jonatack Is this PR really that bad? I spent a few weeks looking at the codebase and tests and I would have raised a similar PR.

Well, we all do. I think the best way for you to answer your question is to review a few dozen of the open PRs. Here are the high-priority ones that really need review: https://github.com/bitcoin/bitcoin/projects/8